### PR TITLE
Explicitly closes WinRmClient and context bus to fix (bad) OS resource leak

### PR DESCRIPTION
**Pull request by nakomis**

### CODE REVIEW

**Comment by maujbs** [comment]
1. Does this fix resolves the problem setting the context (It used to set JSSE context instead of custom context)?
2. Does this change was integrated to master branch? If so, which version?
Thanks!!

**Comment by aledsage** [comment]
@maujbs this fixes a problem introduced in 0.6.0-SNAPSHOT (in commit https://github.com/cloudsoft/winrm4j/commit/c2681198db7aa01d76834d78689149a8d46387da). The cxf bus created in `WinRmClientContext` was not being shutdown.

Which problem setting the context are you referring to? Is there an issue for that?

For the version that this will be in, it will be in 0.6.1. We were in the process of releasing 0.6.0 but then spotted this leak during some longevity testing after 0.6.0 was already in the wild (but before it had been announced). It's an unfortunate situation, but we'll strongly discourage the use of 0.6.0 and release 0.6.1 within the next few days (after doing some more testing).

**Comment by aledsage** [comment]
@nakomis thanks, fix looks good to me. Can you confirm that the longevity testing passes with this? Once that is confirmed, then very happy to merge!

**Comment by maujbs** [comment]
@aledsage thanks. I created an issue a while ago: https://github.com/cloudsoft/winrm4j/issues/85
Example: 
I set the cipher suite because I need FIPS compliance, however I intercepted the request and the cipher suite is the default (JSSE). We need to ensure that the communication has FIPS compliance, but the current implementation leaves behind the cipher suite we set at the beginning.

File: client/src/main/java/io/cloudsoft/winrm4j/client/WinRmClient.java [hunk 1]
```diff
@@ -377,6 +377,15 @@ public void disconnect() {
         }
     }
 
+    @Override
```
Comment (aledsage) [inline client/src/main/java/io/cloudsoft/winrm4j/client/WinRmClient.java]: Can you add some javadoc please. Something like:
```
/**
 * Closes the client. Also closes the associated context if it was created automatically, but not
 * if the context was passed into the builder.
 * 
 * Users should first close all shells that were previously created using this instance's {@link #createShell()}
 * (see {@link ShellCommand#close()}).
 *
 * Do not use a WinRmClient instance after it has been closed.
 */
```

**Comment by maujbs** [comment]
I just saw the fix. Thanks!!
And just to be sure, the fix will be released for the version 0.6.1?

**Pull request merged**